pulp_smash/tests/platform/api_v2/simple_repo_test.py

```diff
@@ -0,0 +1,174 @@
+from pulp_smash.resources.platform.api_v2.api import ERROR_KEYS
+from unittest2 import TestCase
+
+
+class RepoCreateSuccessTestCase(TestCase):
+    """Tests for successfull repo creating functionality.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create repo on pulp server
+        """
+        cls.repo = Repository(id=cls.__name__)
+        cls.repo.create_repo()
+
+    def test_status_code(self):
+        """Test if Create repo returned 201."""
+        self.assertEqual(self.repo.last_response.status_code, 201)
+
+    def test_correct_id(self):
+        """Test if response contain correct repo id.
+        """
+        self.assertEqual(
+            self.repo.last_response.json()['id'],
+            self.__class__.__name__,
+            set(self.repo.last_response.json()))
+
+    @classmethod
+    def tearDownClass(cls):
+        """Delete previously created repository.
+        """
+        cls.repo.delete_repo()
+        cls.repo.last_response.raise_for_status()
+        Task.wait_for_tasks(cls.repo.last_response)
+
+
+class RepoCreateMissingIdTestCase(TestCase):
+    """Tests for repo create functionality with missing required data keys(id).
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create Repository with required data key missing.
+        """
+        cls.repo = Repository(display_name=cls.__name__)
+        cls.repo.create_repo()
+
+    def test_status_code(self):
+        """Check that request returned 400: invalid parameters.
+        """
+        self.assertEqual(self.repo.last_response.status_code, 400,
+                         self.repo.last_response.json())
+
+    def test_body(self):
+        """Test if request returned correct body.
+        """
+        self.assertLessEqual(
+            ERROR_KEYS,
+            set(self.repo.last_response.json().keys()),
+            self.repo.last_response.json())
+
+
+class RepoExistsTestCase(TestCase):
+    """Test if created repo exists on server.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create repository on server with id, description and display_name,
+        test correct status code and get it.
+        """
+        cls.repo = Repository(
+            id=cls.__name__,
+            display_name=cls.__name__,
+            description=cls.__name__
+            )
+        cls.repo.create_repo()
+        cls.repo.last_response.raise_for_status()
+        cls.repo.get_repo()
+
+    def test_status_code(self):
+        """Test if server returned 200.
+        """
+        self.assertEqual(self.repo.last_response.status_code, 200,
+                         self.repo.last_response.json())
+
+    def test_body(self):
+        """Test if repo has all set attributes: id, description and display_name.
+        """
+        self.assertTrue(
+            all(self.repo.last_response.json()[key] == self.__class__.__name__
+                for key in ['id', 'display_name', 'description']))
```

Comment:
This works. But you'll get more useful assertion error messages if you first check that the correct keys are present before checking the value of those keys.

Imagine for a moment that this assertion fails with a `KeyError`, and exception states that the no "display_name" attribute exists. In this case, you know that the server didn't return a "display_name" key, but you're left wondering what other attributes the server returned. Did the server return a "description" key? An assertion like this tells you for sure:

python
self.assertLessEqual(
    {'id', 'display_name', 'description'},
    keys(self.repo.last_response.json())
)
```

You can then go ahead and check the value of each key.